fastly/block_fastly_service_v1_logentries.go

```diff
@@ -0,0 +1,176 @@
+				Default:     20000,
+				Description: "The port number configured in Logentries",
+			},
+			"use_tls": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Whether to use TLS for secure logging",
+			},
+			"format": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "%h %l %u %t %r %>s",
+				Description: "Apache-style string or VCL variables to use for log formatting",
+			},
+			"format_version": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				Description:  "The version of the custom logging format used for the configured endpoint. Can be either 1 or 2. (Default: 1)",
+				ValidateFunc: validateLoggingFormatVersion(),
+			},
+			"response_condition": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "Name of a condition to apply this logging.",
+			},
+			"placement": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Where in the generated VCL the logging call should be placed.",
+				ValidateFunc: validateLoggingPlacement(),
+			},
+		},
+	},
+}
+
+func processLogEntries(d *schema.ResourceData, conn *gofastly.Client, latestVersion int) error {
+	os, ns := d.GetChange("logentries")
+	if os == nil {
+		os = new(schema.Set)
+	}
+	if ns == nil {
+		ns = new(schema.Set)
+	}
+
+	oss := os.(*schema.Set)
+	nss := ns.(*schema.Set)
+	removeLogentries := oss.Difference(nss).List()
+	addLogentries := nss.Difference(oss).List()
+
+	// DELETE old logentries configurations
+	for _, pRaw := range removeLogentries {
+		slf := pRaw.(map[string]interface{})
+		opts := gofastly.DeleteLogentriesInput{
+			Service: d.Id(),
+			Version: latestVersion,
+			Name:    slf["name"].(string),
+		}
+
+		log.Printf("[DEBUG] Fastly Logentries removal opts: %#v", opts)
+		err := conn.DeleteLogentries(&opts)
+		if errRes, ok := err.(*gofastly.HTTPError); ok {
+			if errRes.StatusCode != 404 {
+				return err
+			}
+		} else if err != nil {
+			return err
+		}
+	}
+
+	// POST new/updated Logentries
+	for _, pRaw := range addLogentries {
+		slf := pRaw.(map[string]interface{})
+
+		opts := gofastly.CreateLogentriesInput{
+			Service:           d.Id(),
+			Version:           latestVersion,
+			Name:              slf["name"].(string),
+			Port:              uint(slf["port"].(int)),
+			UseTLS:            gofastly.CBool(slf["use_tls"].(bool)),
+			Token:             slf["token"].(string),
+			Format:            slf["format"].(string),
+			FormatVersion:     uint(slf["format_version"].(int)),
+			ResponseCondition: slf["response_condition"].(string),
+			Placement:         slf["placement"].(string),
+		}
+
+		log.Printf("[DEBUG] Create Logentries Opts: %#v", opts)
+		_, err := conn.CreateLogentries(&opts)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func readLogEntries(conn *gofastly.Client, d *schema.ResourceData, s *gofastly.ServiceDetail) error {
```

Comment:
nitpick - 
```suggestion
func readLogentries(conn *gofastly.Client, d *schema.ResourceData, s *gofastly.ServiceDetail) error {
```

Reply:
This has been updated

Reply:
We've uncommented the ReadElasticSearch and ReadFTP functions as per your PR